Improve description of proces to add a new block device (#133)
ARMmbed/mbed-os-example-filesystem: README.md

## Changing the file system

> In Mbed OS, C++ classes that inherit from the [FileSystem](https://os.mbed.com/docs/latest/reference/storage.html#declaring-a-file-system)

Reviewer: Check the links - they shoudl reference the API in each case:
https://os.mbed.com/docs/mbed-os/latest/apis/filesystem.html
https://os.mbed.com/docs/mbed-os/latest/apis/fatfilesystem.html

> interface are used to implement the file systems. Mbed OS currently supports the [LittleFileSystem](https://os.mbed.com/docs/latest/reference/littlefilesystem.html), and [FATFileSystem](https://os.mbed.com/docs/mbed-os/latest/apis/fatfilesystem.html).  In this example, you can determine which is used, by modifying the declaration of the object ```fs``` in main.cpp. 

Reviewer: ```suggestion
interface are used to implement a file system instance. Mbed OS currently supports the [LittleFileSystem](https://os.mbed.com/docs/latest/reference/littlefilesystem.html), and [FATFileSystem](https://os.mbed.com/docs/mbed-os/latest/apis/fatfilesystem.html).  In this example, you can determine which is used, by modifying the declaration of the object ```fs``` in main.cpp. 
```

For instance, to use the FATFileSystem, declare ```fs``` as an instance of the class FATFileSystem: 

> ``` cpp

Reviewer: That's okay but I prefer the original diff.

blocks. You can find the number of blocks on a block device by dividing the
block device's size by its erase size.

## Changing the block device

> Mbed-OS supports a variety of [block devices](https://os.mbed.com/docs/mbed-os/v6.5/apis/data-storage.html#Default-BlockDevice-configuration). You can find information on configuring, and changing the block device used in this example [here](https://github.com/ARMmbed/mbed-os-example-blockdevice#changing-the-block-device).

Reviewer: The link uses "v6.5" instead of "latest"

blocks. You can find the number of blocks on a block device by dividing the
block device's size by its erase size.

> Mbed OS has two filesystem options:

Author: Similar to the block device section can we link to the mbed os documentation rather than duplicating the content here?

> Mbed OS has two filesystem options:

Reviewer: Agreed

## Changing the block device

> Mbed-OS supports a variety of block device types, more information on supported devices can be found [here](https://os.mbed.com/docs/mbed-os/v6.5/apis/data-storage.html#Default-BlockDevice-configuration). Each device is represented by a C++ class that inherits from the interface [BlockDevice](https://os.mbed.com/docs/latest/reference/storage.html#block-devices). A block device may be added by either modifying one of the component or applications configuration files, as well as, by re-implementing `BlockDevice::get_default_instance()` to bypass the default settings. This process is described in greater detail [here](https://github.com/ARMmbed/mbed-os-example-blockdevice#changing-the-block-device).

Reviewer: ```suggestion
Mbed-OS supports a variety of [block devices](https://os.mbed.com/docs/mbed-os/latest/apis/data-storage.html#Default-BlockDevice-configuration). You can find information on configuring and changing the block device used by the example [here](https://github.com/ARMmbed/mbed-os-example-blockdevice#changing-the-block-device).
```
The information that you added on how to change the block device should go in the other page.
